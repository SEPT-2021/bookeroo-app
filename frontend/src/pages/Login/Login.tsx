import React, { useState } from "react";
import axios from "axios";

import {
  Avatar,
  Box,
  Button,
  Checkbox,
  createStyles,
  FormControlLabel,
  Grid,
  Hidden,
  Link,
  Paper,
  TextField,
  Theme,
  Typography,
  withStyles,
  WithStyles,
} from "@material-ui/core";
import { LockOutlined } from "@material-ui/icons";
import logo from "../../assets/logo.svg";

function Login({ classes }: LoginProps) {
  const [Username, setUsername] = useState("");
  const [Password, setPassword] = useState("");

  const loginDetails = {
    username: Username,
    password: Password,
  };

  function loginPost() {
    return axios({
      method: "POST",
      url: "/api/users/login",
      data: { loginDetails },
    })
      .then((response) => {
        // eslint-disable-next-line no-console
        console.log(response.data);
      })
      .catch((error) => {
        if (error) {
          // eslint-disable-next-line no-console
          console.log(error.message);
        }
      });
  }

  async function getLoginPostResponse() {
    return loginPost();
  }

  return (
    <Grid container component="main" className={classes.root}>
      <Grid item xs={12} sm={4} md={7} className={classes.steps}>
        <Box
          display="flex"
          alignItems="center"
          justifyContent="center"
          flexWrap="wrap"
          width="100%"
        >
          <img src={logo} alt="logo" />
          <Hidden xsDown>
            <div>
              <Typography variant="h1">Bookeroo</Typography>
              <Typography variant="h5">
                Dealing books has never been this easy
              </Typography>
            </div>
          </Hidden>
        </Box>
      </Grid>
      <Grid item xs={12} sm={8} md={5} component={Paper} elevation={6} square>
        <div className={classes.paper}>
          <Avatar className={classes.avatar}>
            <LockOutlined />
          </Avatar>
          <Typography component="h1" variant="h5">
            Sign in
          </Typography>
          <Box>
            <TextField
              variant="outlined"
              margin="normal"
              required
              fullWidth
              id="email"
              label="Email Address"
              name="email"
              autoComplete="email"
              autoFocus
              onChange={(e) => setUsername(e.target.value)}
            />
            <TextField
              variant="outlined"
              margin="normal"
              required
              fullWidth
              name="password"
              label="Password"
              type="password"
              id="password"
              autoComplete="current-password"
              onChange={(e) => setPassword(e.target.value)}
            />
            <FormControlLabel
              control={<Checkbox value="remember" color="primary" />}
              label="Remember me"
            />
            <Button
              type="submit"
              fullWidth
              variant="contained"
              color="primary"
              className={classes.submit}
              onClick={getLoginPostResponse}
            >
              Sign In
            </Button>
            <Grid container>
              <Grid item xs>
                <Link href="/forgot-password" variant="body2">
                  Forgot password?
                </Link>
              </Grid>
              <Grid item>
                <Link href="/api/users/register" variant="body2">
                  Don't have an account? Register
                </Link>
              </Grid>
            </Grid>
            <Box mt={5}>
              <Typography variant="caption">
                Disclaimer: This application was built for SEPT 2021, and is a
                university project. Please do not provide valuable data into
                this application.
              </Typography>
            </Box>
          </Box>
        </div>
      </Grid>
    </Grid>
  );
}

const styles = (theme: Theme) =>
  createStyles({
    root: {
      height: "100vh",
    },
    steps: {
      "& img": {
        width: 230,
        margin: theme.spacing(6),
      },
      display: "flex",
      alignItems: "center",
      justifyContent: "center",
      flexDirection: "column",
    },
    paper: {
      margin: theme.spacing(8, 4),
      display: "flex",
      flexDirection: "column",
      alignItems: "center",
    },
    avatar: {
      margin: theme.spacing(1),
      backgroundColor: theme.palette.secondary.main,
    },
    submit: {
      margin: theme.spacing(3, 0, 2),
    },
  });

type LoginProps = WithStyles<typeof styles>;

export default withStyles(styles)(Login);
