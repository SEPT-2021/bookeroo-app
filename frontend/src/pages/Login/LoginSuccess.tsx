import React from "react";

import {
  createStyles,
  Grid,
  Theme,
  withStyles,
  WithStyles,
} from "@material-ui/core";

function LoginSuccess({ classes }: LoginSuccessProps) {
  return (
    <Grid className={classes.root}>
      <h1> LOGIN SUCCESS </h1>
    </Grid>
  );
}

const styles = (theme: Theme) =>
  createStyles({
    root: {
      height: "100vh",
    },
    steps: {
      "& img": {
        width: 230,
        margin: theme.spacing(6),
      },
      display: "flex",
      alignItems: "center",
      justifyContent: "center",
      flexDirection: "column",
    },
    paper: {
      margin: theme.spacing(8, 4),
      display: "flex",
      flexDirection: "column",
      alignItems: "center",
    },
    avatar: {
      margin: theme.spacing(1),
      backgroundColor: theme.palette.secondary.main,
    },
    submit: {
      margin: theme.spacing(3, 0, 2),
    },
  });

type LoginSuccessProps = WithStyles<typeof styles>;

export default withStyles(styles)(LoginSuccess);
