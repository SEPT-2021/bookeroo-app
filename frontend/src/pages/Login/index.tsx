import React, { useState } from "react";
import { Redirect } from "react-router-dom";
import { LockOutlined } from "@material-ui/icons";
import { useMutation } from "react-query";
import {
  Avatar,
  Box,
  createStyles,
  Grid,
  Hidden,
  Paper,
  Theme,
  Typography,
  withStyles,
  WithStyles,
} from "@material-ui/core";
import logo from "../../assets/logo.svg";
import Link from "../../util/Link";
import { loginUser } from "../../util/api";
import LoadingButton from "../../util/LoadingButton";
import FormField from "../../util/FormField";

function Login({ classes }: LoginProps) {
  const [username, setUsername] = useState("");
  const [password, setPassword] = useState("");

  const { isLoading, mutate, error, data, isSuccess } = useMutation(loginUser);
  const onSubmit = () => mutate({ username, password });
  // TODO testing
  // eslint-disable-next-line no-console
  const ls = localStorage.getItem("token");
  // eslint-disable-next-line no-console
  console.log(ls);

  if (isSuccess) {
    localStorage.setItem("success", String(data?.success as boolean));
    localStorage.setItem("token", data?.token as string);

    return <Redirect to="/loginSuccess" />;
  }

  return (
    <Grid container component="main" className={classes.root}>
      <Grid item xs={12} sm={4} md={7} className={classes.steps}>
        <Box
          display="flex"
          alignItems="center"
          justifyContent="center"
          flexWrap="wrap"
          width="100%"
        >
          <img src={logo} alt="logo" />
          <Hidden xsDown>
            <div>
              <Typography variant="h1">Bookeroo</Typography>
              <Typography variant="h5">
                Dealing books has never been this easy
              </Typography>
            </div>
          </Hidden>
        </Box>
      </Grid>
      <Grid item xs={12} sm={8} md={5} component={Paper} elevation={6} square>
        <div className={classes.paper}>
          <Avatar className={classes.avatar}>
            <LockOutlined />
          </Avatar>
          <Typography component="h1" variant="h5">
            Sign in
          </Typography>
          <Box>
            <FormField
              errors={error?.response?.data}
              name="username"
              autoComplete="email"
              label="Email Address"
              value={username}
              onChange={setUsername}
            />
            <FormField
              errors={error?.response?.data}
              name="password"
              label="Password"
              type="password"
              autoComplete="current-password"
              onChange={setPassword}
            />
            <LoadingButton
              loading={isLoading}
              fullWidth
              variant="contained"
              color="primary"
              className={classes.submit}
              onClick={onSubmit}
            >
              Sign In
            </LoadingButton>
            <Grid container>
              <Grid item xs>
                <Link to="/forgot-password" variant="body2">
                  Forgot password?
                </Link>
              </Grid>
              <Grid item>
                <Link to="/register" variant="body2">
                  Don't have an account? Register
                </Link>
              </Grid>
            </Grid>
            <Box mt={5}>
              <Typography variant="caption">
                Disclaimer: This application was built for SEPT 2021, and is a
                university project. Please do not provide valuable data into
                this application.
              </Typography>
            </Box>
          </Box>
        </div>
      </Grid>
    </Grid>
  );
}

const styles = (theme: Theme) =>
  createStyles({
    root: {
      height: "100vh",
    },
    steps: {
      "& img": {
        width: 230,
        margin: theme.spacing(6),
      },
      display: "flex",
      alignItems: "center",
      justifyContent: "center",
      flexDirection: "column",
    },
    paper: {
      margin: theme.spacing(15, 4),
      [theme.breakpoints.down("sm")]: {
        margin: theme.spacing(8, 4),
      },
      display: "flex",
      flexDirection: "column",
      alignItems: "center",
    },
    avatar: {
      margin: theme.spacing(1),
      backgroundColor: theme.palette.secondary.main,
    },
    submit: {
      margin: theme.spacing(3, 0, 2),
    },
  });

type LoginProps = WithStyles<typeof styles>;

export default withStyles(styles)(Login);
