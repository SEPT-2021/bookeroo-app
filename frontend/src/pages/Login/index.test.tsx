import React from "react";
import { render } from "@testing-library/react";
import { QueryClient, QueryClientProvider } from "react-query";
import Login from ".";

jest.mock("../../util/Link", () => "mock-Link");
jest.mock("../../util/LoadingButton", () => "mock-LoadingButton");
jest.mock("../../util/FormField", () => "mock-FormField");

test("renders login", () => {
  const { container } = render(
    <QueryClientProvider client={new QueryClient()}>
      <Login />
    </QueryClientProvider>
  );
  expect(container).toMatchSnapshot();
});
