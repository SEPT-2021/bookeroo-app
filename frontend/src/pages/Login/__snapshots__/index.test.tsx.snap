// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders login 1`] = `
<div>
  <main
    class="MuiGrid-root Login-root-1 MuiGrid-container"
  >
    <div
      class="MuiGrid-root Login-steps-2 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-7"
    >
      <div
        class="MuiBox-root MuiBox-root-6"
      >
        <img
          alt="logo"
          src="logo.svg"
        />
      </div>
    </div>
    <div
      class="MuiPaper-root MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-8 MuiGrid-grid-md-5 MuiPaper-elevation6"
    >
      <div
        class="Login-paper-3"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular Login-avatar-4 MuiAvatar-colorDefault"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
            />
          </svg>
        </div>
        <h1
          class="MuiTypography-root MuiTypography-h5"
        >
          Sign in
        </h1>
        <div
          class="MuiBox-root MuiBox-root-7"
        >
          <mock-formfield
            autocomplete="email"
            label="Email Address"
            name="username"
            value=""
          />
          <mock-formfield
            autocomplete="current-password"
            label="Password"
            name="password"
            type="password"
          />
          <mock-loadingbutton
            classname="Login-submit-5"
            color="primary"
            fullwidth="true"
            loading="false"
            variant="contained"
          >
            Sign In
          </mock-loadingbutton>
          <div
            class="MuiGrid-root MuiGrid-container"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
            >
              <mock-link
                to="/forgot-password"
                variant="body2"
              >
                Forgot password?
              </mock-link>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <mock-link
                to="/register"
                variant="body2"
              >
                Don't have an account? Register
              </mock-link>
            </div>
          </div>
          <div
            class="MuiBox-root MuiBox-root-8"
          >
            <span
              class="MuiTypography-root MuiTypography-caption"
            >
              Disclaimer: This application was built for SEPT 2021, and is a university project. Please do not provide valuable data into this application.
            </span>
          </div>
        </div>
      </div>
    </div>
  </main>
</div>
`;
